search-commons/src/main/java/no/unit/nva/search/IndexDocument.java

```diff
@@ -204,146 +76,44 @@ public boolean equals(Object o) {
-            return this;
-        }
-
-        public Builder withId(SortableIdentifier id) {
-            this.id = id;
-            return this;
-        }
-
-        public Builder withDoi(URI doi) {
-            this.doi = doi;
-            return this;
-        }
-
-        public Builder withOwner(String owner) {
-            this.owner = owner;
-            return this;
-        }
-
-        public Builder withContributors(List<IndexContributor> contributors) {
-            this.contributors = contributors;
-            return this;
-        }
-
-        public Builder withTitle(String title) {
-            this.title = title;
-            return this;
-        }
-
-        public Builder withDescription(String description) {
-            this.description = description;
-            return this;
-        }
-
-        public Builder withAbstract(String publicationAbstract) {
-            this.publicationAbstract = publicationAbstract;
-            return this;
-        }
-
-        @JacocoGenerated
-        @SuppressWarnings("PMD.NullAssignment")
-        public Builder withPublicationDate(IndexDate date) {
-            this.publicationDate = isNonNullDate(date) ? date : null;
-            return this;
-        }
-
-        public Builder withPublisher(IndexPublisher publisher) {
-            this.publisher = publisher;
-            return this;
-        }
-
-        public Builder withModifiedDate(Instant modifiedDate) {
-            this.modifiedDate = modifiedDate;
-            return this;
-        }
-
-        public Builder withPublishedDate(Instant publishedDate) {
-            this.publishedDate = publishedDate;
-            return this;
-        }
-
-        public Builder withAlternativeTitles(Map<String, String> alternativeTitles) {
-            if (nonNull(alternativeTitles)) {
-                this.alternativeTitles = Map.copyOf(alternativeTitles);
-            } else {
-                this.alternativeTitles = Collections.emptyMap();
-            }
-            return this;
-        }
-
-        public Builder withTags(List<String> tags) {
-            if (nonNull(tags)) {
-                this.tags = List.copyOf(tags);
-            } else {
-                this.tags = Collections.emptyList();
-            }
-            return this;
-        }
-
-        public Builder withReference(Reference reference) {
-            if (nonNull(reference)) {
-                this.reference = reference;
-            }
-            return this;
-        }
-
-        public IndexDocument build() {
-            return new IndexDocument(this);
+    /**
+     * JsonString.
+     *
+     * @return JsonString
+     */
+    @Override
+    @JacocoGenerated
+    public String toJsonString() {
+        try {
+            return objectMapper.writeValueAsString(addContext(root));
+        } catch (JsonProcessingException e) {
+            logger.error(PROBLEM_SERIALIZING_MESSAGE, e);
+            return EMPTY_JSON_OBJECT;
```

Comment:
if something is not serializable we just return empty object? What is the possibility of not being able to serialize a JsonNode?

Reply:
The possibillity is next to nothing, but the consequence is fatal....